src/webhook/upgrade_operator/fetch_upgradeconfig.py

```diff
@@ -0,0 +1,40 @@
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from kubernetes.config.config_exception import ConfigException
+
+
+def kube_auth():
+    # Use ~/.kube/config for authentication if script run directly
+    if __name__ == "__main__":
+        try:
+            config.load_kube_config()
+            return True
+        except ConfigException as c:
+            print("ConfigException in load_kube_config() : %s\n" % c)
+            return False
+
+    # Use serviceaccount token to authenticate.
+    else:
```

Comment:
Note that this `else` is redundant, because the code under the `if` always returns.

Stylistically, though, it's not a terrible thing to keep it.